### ISSUE TITLE
[reportCallIssue] False positive with metaclass

### ISSUE THREAD

**Issue by last-partizan** [microsoft/pyright#7437]
**Describe the bug**
This is stripped down code from `django-admin-autocomplete-filter `, which triggers this error:

**Code or Screenshots**

```python
class AutocompleteFilter:
    def __init__(self, request, params, model, model_admin):
        pass


def AutocompleteFilterFactory():
    class NewMetaFilter(type(AutocompleteFilter)):
        def __new__(cls, name, bases, attrs):
            ...

    class NewFilter(AutocompleteFilter, metaclass=NewMetaFilter):

        def __init__(self, request, params, model, model_admin):
            super().__init__(request, params, model, model_admin)
            #                ^^^^^^^ error: Expected 0 positional arguments (reportCallIssue)

    return NewFilter
```

**VS Code extension or command-line**

```
pyright 1.1.353
```

**Comment by erictraut** [comment]
This results from `type(AutocompleteFilter) ` evaluating to `Any`. That means `NewMetaFilter` is a class that derives from `Any`. Once you introduce an `Any` into a class MRO, many things can go squirrelly with type analysis. 

In this particular case, a type checker should be able to determine a better type for `type(AutocompleteFilter)`. It's simply the metaclass for `AutocompleteFilter`, which is `type`. 

I've updated pyright to do this, which eliminates the error.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.354, which I just published.
